python/ray/data/tests/test_streaming_integration.py

```diff
@@ -109,35 +123,39 @@ def get_lengths(*iterators, use_iter_batches=True):
         i1,
         i2,
     ) = ds.streaming_split(2, equal=True)
-    lengths = get_lengths(i1, i2)
-    assert lengths == [500, 500], lengths
+    for _ in range(2):
+        lengths = get_lengths(i1, i2)
+        assert lengths == [500, 500], lengths
 
     ds = ray.data.range(1)
     (
         i1,
         i2,
     ) = ds.streaming_split(2, equal=True)
-    lengths = get_lengths(i1, i2)
-    assert lengths == [0, 0], lengths
+    for _ in range(2):
+        lengths = get_lengths(i1, i2)
+        assert lengths == [0, 0], lengths
 
     ds = ray.data.range(1)
     (
         i1,
         i2,
     ) = ds.streaming_split(2, equal=False)
-    lengths = get_lengths(i1, i2)
-    assert lengths == [0, 1], lengths
+    for _ in range(2):
+        lengths = get_lengths(i1, i2)
+        assert lengths == [0, 1], lengths
 
     ds = ray.data.range(1000, parallelism=10)
     for equal_split, use_iter_batches in itertools.product(
         [True, False], [True, False]
     ):
         i1, i2, i3 = ds.streaming_split(3, equal=equal_split)
-        lengths = get_lengths(i1, i2, i3, use_iter_batches=use_iter_batches)
-        if equal_split:
-            assert lengths == [333, 333, 333], lengths
-        else:
-            assert lengths == [300, 300, 400], lengths
+        for _ in range(2):
+            lengths = get_lengths(i1, i2, i3, use_iter_batches=use_iter_batches)
+            if equal_split:
+                assert lengths == [333, 333, 333], lengths
+            else:
+                assert lengths == [300, 300, 400], lengths
```

Comment:
add a test for the barrier case? simulating one consumer iterating through the  epoch faster than the other.

Reply:
Good suggestion. There is actually a bug here if iterators are copied.